artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/AbstractJournalStorageManager.java

```diff
@@ -1481,6 +1509,7 @@ public JournalLoadInformation loadBindingJournal(final List<QueueBindingInfo> qu
          } else if (rec == JournalRecordIds.ADDRESS_BINDING_RECORD) {
             PersistentAddressBindingEncoding bindingEncoding = newAddressBindingEncoding(id, buffer);
             addressBindingInfos.add(bindingEncoding);
+            mapAddressBindings.put(id, bindingEncoding);
          } else if (rec == JournalRecordIds.GROUP_RECORD) {
```

Comment:
Is the a compatibility test to ensure broker upgrade downgrade there is no issues with bindings. E.g. old broker can read new binding files ok, and new broker version can read old generated binding files.

Reply:
Is there such a test already existing somewhere ?

Reply:
Yes theres a whole compatibility test suite.

Reply:
@clebertsuconic will be best to chat to

Reply:
ok I wrote some compatibility tests